corehq/apps/registration/utils.py

```diff
@@ -98,7 +98,11 @@ def is_user_subscribed_with_email(couch_user):
                 and couch_user.email == old_email)
     users_subscribed_with_old_email = [
         other_user
-        for other_user in CouchUser.all()
+        for other_user in CouchUser.view(
+            'users/mailing_list_emails',
```

Comment:
You should add `key=old_email` to restrict the results to only those users with `email = old_email`.